cmd/spiderpool-init/cmd/client.go

```diff
@@ -183,39 +183,37 @@ func (c *CoreClient) WaitMultusCNIConfigCreated(ctx context.Context, multuscnico
 
 func (c *CoreClient) WaitPodListReady(ctx context.Context, namespace string, labels map[string]string) error {
 	logger := logutils.FromContext(ctx)
+	ticker := time.NewTicker(retryIntervalSec * time.Second)
+	defer ticker.Stop()
 
 	var podList corev1.PodList
-	var err error
-	noReady := true
-	interval := retryIntervalSec * time.Second
-	for noReady {
-		if err = c.List(ctx, &podList, client.MatchingLabels(labels), client.InNamespace(namespace)); err != nil {
-			logger.Sugar().Errorf("failed to get spiderAgent pods: %v, retrying...", err)
-			time.Sleep(interval)
-			continue
-		}
+	for {
+		select {
+		case <-ticker.C:
+			if err := c.List(ctx, &podList, client.MatchingLabels(labels), client.InNamespace(namespace)); err != nil {
```

Comment:
I remember the ctx is avaivable just for 5 mins ? after that ,this will fail

Reply:
I think this job should be a  afterInstall hook things of helm, and that makes more sense for installation and timeout